risc0/zkvm/src/receipt/segment.rs

```diff
@@ -107,6 +120,33 @@ impl SegmentReceipt {
     }
 }
 
+/// Verifier parameters used to verify a [SegmentReceipt].
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct SegmentReceiptVerifierParameters {
+    /// Set of control ID with which the receipt is expected to verify.
+    pub control_ids: BTreeSet<Digest>,
```

Comment:
I wonder if this will be good for future deserializers. `BTreeSet` is a nice type, but `Vec` might be more future compatible. Not sure what the right thing here is.

Reply:
Good question. I'm not sure what is best here either. On serialization, at least `serde` treats `BTreeSet` as a generic sequence [1]. I was curious, so I tried decoding an encoded `BTreeSet` into a `Vec` and that also worked [2]. Doesn't fully answer the question though.

[1] https://docs.rs/serde/latest/src/serde/ser/impls.rs.html#239-243
[2] https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=437b6972c4d482f7f024e76c5d12a69c